UPGRADE.md

```diff
@@ -1175,6 +1175,290 @@ const mapStateToProps = state => ({
 export default connect(mapStateToProps)(Menu);
 ```
 
+## Logout is now displayed in the AppBar on desktop
+
+The Logout button is now displayed in the AppBar on desktop but is still displayed as a menu item on small devices.
+
+This impacts how you build a custom menu, as you'll now have to check whether you are on small devices before displaying the logout:
+
+```jsx
+// in src/Menu.js
+// before
+import React from 'react';
+import { connect } from 'react-redux';
+import { MenuItemLink, getResources } from 'react-admin';
+import { withRouter } from 'react-router-dom';
+
+const Menu = ({ resources, onMenuClick, logout }) => (
+    <div>
+        {resources.map(resource => (
+            <MenuItemLink to={`/${resource.name}`} primaryText={resource.name} onClick={onMenuClick} />
+        ))}
+        <MenuItemLink to="/custom-route" primaryText="Miscellaneous" onClick={onMenuClick} />
+        {logout}
+    </div>
+);
+
+const mapStateToProps = state => ({
+    resources: getResources(state),
+});
+
+export default withRouter(connect(mapStateToProps)(Menu));
+
+// after
+import React from 'react';
+import { connect } from 'react-redux';
+import { MenuItemLink, getResources } from 'react-admin';
+import { withRouter } from 'react-router-dom';
+import Responsive from '../layout/Responsive';
+
+const Menu = ({ resources, onMenuClick, logout }) => (
+    <div>
+        {resources.map(resource => (
+            <MenuItemLink to={`/${resource.name}`} primaryText={resource.name} onClick={onMenuClick} />
+        ))}
+        <MenuItemLink to="/custom-route" primaryText="Miscellaneous" onClick={onMenuClick} />
+        <Responsive
+            small={logout}
+            medium={<div />} // We must define something to not fallback on small
+        />
+    </div>
+);
+
+const mapStateToProps = state => ({
+    resources: getResources(state),
+});
+
+export default withRouter(connect(mapStateToProps)(Menu));
+```
+
+It also impacts custom layouts if you're using the default `AppBar` in it. You now have to pass the `logout` prop to the `AppBar`:
+
+```jsx
+// in src/MyLayout.js
+// Before
+import React, { createElement, Component } from 'react';
```

Comment:
Does all the code concern the `logout` ?  Because otherwise it's hard to see what changes affect the `logout`.

Reply:
I just committed a simplified version, is it better ?

Reply:
:+1: